src/js/components/modals/AppModalComponent.jsx

```diff
@@ -165,11 +175,15 @@ var AppModalComponent = React.createClass({
 
     // Ports should always be an Array.
     if ("ports" in modelAttrs) {
-      var portStrings = modelAttrs.ports.split(",");
-      modelAttrs.ports = portStrings.map(function (p) {
-        var port = parseInt(p, 10);
-        return isNaN(port) ? p : port;
-      });
+      if (modelAttrs.ports === "") {
```

Comment:
I am also unhappy with the structure and want to refactor the whole data handling at some point to make it better understandable.

Especially this particular case here is a little bit tricky.

Some explanation:
1. 
```modelAttrs.ports = null``` means that this attribute will be untouched on the model, so ```model.ports = 1000``` won't be overriden.
2. 
```modelAttrs.ports = ""``` that the user wants to reset the values to nothing/default.
3.
```modelAttrs.ports.split(",");``` needs an array, otherwise it crashes.

These conditions lead to this nested structure.